src/TestFramework/Coverage/JUnit/JUnitTestCaseSorter.php

```diff
@@ -98,9 +156,11 @@ private function uniqueByTestFile(array $testLocations): array
         $uniqueTests = [];
 
         foreach ($testLocations as $testLocation) {
+            /** @var string $filePath */
```

Comment:
on theory there is no guarantee that this is non-null, and even if is it won't mess up the code bellow either, so can be removed?

Reply:
Path for TestFileTimeData is not optional. It is never a null.